helpers/create_pnp_backdata.py

```diff
@@ -0,0 +1,540 @@
+    within backdata_schema.toml.
+
+    Args:
+        df (pd.DataFrame): The dataframe to convert the column datatypes.
+    Return:
+        DF (pd.DataFrame): The dataframe with the converted column datatypes.
+
+    """
+    with open('./config/backdata_schema.toml', 'r') as f:
+        datatypes_config = toml.load(f)
+
+    datatypes_dict = {}
+
+    for column, datatype in datatypes_config.items():
+        datatypes_dict.update({column: datatype["Deduced_Data_Type"]})
+
+    for i in list(df.columns):
+        if i in list(datatypes_dict.keys()):
+            try:
+                df[i] = df[i].astype(datatypes_dict[i])
+            except:
+                pass
+        else:
+            continue
+
+    return df
+
+
+def get_status_encoded(df):
+    """ Function to populate the statusencoded column based on the mapping of the
+    'map cora status to column "status"' column.
+
+    Args:
+        df (pd.DataFrame): The dataframe to populate the statusencoded column.
+    Return:
+        DF (pd.DataFrame): The dataframe with the populated statusencoded column.
+    """
+    cora_to_status_encoded_dict = {200: 100,
+                                   100: 101,
+                                   1000: 102,
+                                   400: 200,
+                                   500: 201,
+                                   600: 210,
+                                   800: 211,
+                                   1200: 302,
+                                   1300: 303,
+                                   900: 304,
+                                   1400: 309}
+
+    df["statusencoded"] = df['map cora status to column "status"'].map(
+        cora_to_status_encoded_dict
+    )
+
+    return df
+
+
+def identify_key_business(df):
+    """ Function to identify the key business using reference column & key
+    busineses lookup table.
+
+    Args:
+        df (pd.DataFrame): The dataframe to identify the key business columns.
+    Return:
+        df (pd.DataFrame): The dataframe with the identified key business columns.
+    """
+    # get key businesses
+    key_businesses_df = pd.read_csv(os.path.join(rootpath, 'KEYS 2023.csv'))
+    key_businesses_list = list(key_businesses_df["2023 KEYS"])
+
+    df['pnp_key'] = df['reference'].apply(
+        lambda x: 'key0' if x in key_businesses_list else 'key1'
+    )
+
+    return df
+
+
+def identify_osmotherly_businesses(df):
+    """ Function to identify the osmotherly businesses using reference
+    column & osmotherly busineses lookup table.
+
+    Args:
+        df (pd.DataFrame): The dataframe to identify the osmotherly business
+        columns.
+    Return:
+        df (pd.DataFrame): The dataframe with the identified osmotherly business
+        columns.
+    """
+    # get osmotherly businesses
+    osmotherly_businesses_df = pd.read_csv(os.path.join(rootpath,
+                                                        "Osmotherly PNP 2023.csv"))

```

Comment:
Can you make sure that next time you do a commit, the pre-commit formatting works for all of this script?
We use a different system for continuation over lines, as you know.

Reply:
I wonder if my pre-commit was working at the time I committed this work.